### ISSUE TITLE
linter: `node/no-exports-assign` should be `correctness` or `suspicious` category

### ISSUE THREAD

**Issue by sapphi-red** [oxc-project/oxc#26282]
[`node/no-exports-assign`](https://oxc.rs/docs/guide/usage/linter/rules/node/no-exports-assign.html) is currently in `style` category.

But I think this should be `correctness` or `suspicious`.
```js
exports = {
  foo: 1
}
```
does nothing. This isn't a matter of preference. It has to be
```js
module.exports = {
  foo: 1
}
```
or
```js
exports.foo = 1
```
to expose `foo`.

related docs: https://nodejs.org/docs/latest-v24.x/api/modules.html#exports-shortcut
